src/bootstrap/bootstrap.py

```diff
@@ -180,6 +180,7 @@ def parse_nightly_dates(self):
 
     def build_bootstrap(self):
         env = os.environ.copy()
+        env["CARGO_HOME"] = os.path.join(self.build_dir, "cargo")
```

Comment:
Let's leave this out for now as it seems unrelated to this change, and this can always be set externally or we can add a configuration option for it.